lib/Sema/TypeCheckAttr.cpp

```diff
@@ -2549,26 +2524,40 @@ void AttributeChecker::visitDifferentiableAttr(DifferentiableAttr *attr) {
   }
 
   // Resolve the adjoint declaration.
+  FuncDecl *resolvedAdjoint = nullptr;
+  auto adjointSpecifier = attr->getAdjoint();
+
+  auto adjointTypeCtx = original->getInnermostTypeContext();
+  if (!adjointTypeCtx) adjointTypeCtx = original->getParent();
+
+  Diagnostic adjointOverloadDiagnostic(
```

Comment:
As we discussed in person, you can turn these into thunks.

Reply:
Done.